test/1.4/lib/T_internal.dml

```diff
@@ -0,0 +1,49 @@
+/*
+  © 2021 Intel Corporation
+  SPDX-License-Identifier: MPL-2.0
+*/
+dml 1.4;
+
+device test;
+
+import "simics/simulator-api.dml";
+import "testing.dml";
+
+bank b {
+    param register_size = 1;
+    register r0 @ 0;
+    register r1 @ 1 "short desc";
+    register r2 @ 2 {
+        param documentation = "long desc";
+    }
+    register r3 @ 3 {
+        param internal = false;
+    }
+}
+
+attribute a0 is bool_attr;
+attribute a1 is bool_attr "short desc";
+attribute a2 is bool_attr {
+    param documentation = "long desc";
+}
+attribute a3 is bool_attr {
+    param internal = false;
+}
+
+method get_internal(const char *cls, const char *attr) -> (bool) {
+    return (SIM_get_attribute_attributes(SIM_get_class(cls), attr)
+            & Sim_Attr_Internal) == Sim_Attr_Internal;
```

Comment:
You can use `SIM_object_class` together with the `obj` param of the device object or the banks instead. I think that's cleaner.